api/envoy/config/common/mutation_rules/v3/mutation_rules.proto

```diff
@@ -84,3 +85,19 @@ message HeaderMutationRules {
   // Default is false.
   google.protobuf.BoolValue disallow_is_error = 7;
 }
+
+// HeaderMutation defines an API to apply match and rewrite operations to header value.
+message HeaderMutation {
+  // The header name of the request header that will be mutated.
+  string header = 1
+      [(validate.rules).string = {min_len: 1 well_known_regex: HTTP_HEADER_NAME strict: false}];
+
+  // Optional new header name. If not empty, the final header name after mutation will be replaced with this name.
+  string rename = 2 [
+    (validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false ignore_empty: true}
+  ];
+
+  // Regex match and rewrite operations that will be applied to the header value.
+  type.matcher.v3.RegexMatchAndSubstitute regex_rewrite = 3
+      [(validate.rules).message = {required: true}];
```

Comment:
I think it might be cleaner to have this inside of a oneof so it's clearly one operation per mutation? And then make the oneof required to be set via PGV? wdyt?

Reply:
Sorry, may be my comment is not clear enough.
`rename` is used if we just want to set a new header with the value from the regex substitution (with the original header value as input) and key from the `rename`.
Both original header and new header will be kept.

Seems `oneof` couldn't achieve this target.